libs/backend/src/cast_vote_records/import.test.ts

```diff
@@ -1,7 +1,325 @@
+    arePollsClosed: true,
+    castVoteRecordReportMetadata: expect.any(Object),
+    castVoteRecordRootHash: expect.stringMatching(/^[a-f0-9]{64}$/),
+  });
+
+  let encounteredReferencedImageFiles = false;
+  let encounteredReferencedLayoutFiles = false;
+  for await (const castVoteRecordResult of castVoteRecordIterator) {
+    expect(castVoteRecordResult.isOk()).toEqual(true);
+    assert(castVoteRecordResult.isOk());
+    const { castVoteRecord, referencedFiles } = castVoteRecordResult.ok();
+    expect(castVoteRecord).toEqual(expect.any(Object));
+
+    if (referencedFiles) {
+      encounteredReferencedImageFiles = true;
+      for (const i of [0, 1] as const) {
+        const imageFileReadResult = await referencedFiles.imageFiles[i].read();
+        expect(imageFileReadResult.isOk()).toEqual(true);
+        assert(imageFileReadResult.isOk());
+        const image = imageFileReadResult.ok();
+        expect(image).toEqual(expect.any(Buffer));
+
+        if (referencedFiles.layoutFiles) {
+          encounteredReferencedLayoutFiles = true;
+          const layoutFileReadResult =
+            await referencedFiles.layoutFiles[i].read();
+          expect(layoutFileReadResult.isOk()).toEqual(true);
+          assert(layoutFileReadResult.isOk());
+          const layout = layoutFileReadResult.ok();
+          expect(layout).toEqual(expect.any(Object));
+        }
+      }
+    }
+  }
+  expect(encounteredReferencedImageFiles).toEqual(true);
+  expect(encounteredReferencedLayoutFiles).toEqual(true);
+});
+
+test('authentication error during import', async () => {
+  const exportDirectoryPath = castVoteRecordExport.asDirectoryPath();
+  fs.appendFileSync(
+    path.join(exportDirectoryPath, CastVoteRecordExportFileName.METADATA),
+    '\n'
+  );
+
+  expect(await readCastVoteRecordExport(exportDirectoryPath)).toEqual(
+    err({ type: 'authentication-error' })
+  );
+});
+
+test('metadata file not found during import', async () => {
+  mockFeatureFlagger.enableFeatureFlag(
+    BooleanEnvironmentVariableName.SKIP_CAST_VOTE_RECORDS_AUTHENTICATION
+  );
+  const exportDirectoryPath = castVoteRecordExport.asDirectoryPath();
+  fs.rmSync(
+    path.join(exportDirectoryPath, CastVoteRecordExportFileName.METADATA)
+  );
+
+  expect(await readCastVoteRecordExport(exportDirectoryPath)).toEqual(
+    err({ type: 'metadata-file-not-found' })
+  );
+});
+
+test('metadata file parse error during import', async () => {
+  mockFeatureFlagger.enableFeatureFlag(
+    BooleanEnvironmentVariableName.SKIP_CAST_VOTE_RECORDS_AUTHENTICATION
+  );
+  const exportDirectoryPath = castVoteRecordExport.asDirectoryPath();
+  fs.appendFileSync(
+    path.join(exportDirectoryPath, CastVoteRecordExportFileName.METADATA),
+    '}'
+  );
+
+  expect(await readCastVoteRecordExport(exportDirectoryPath)).toEqual(
+    err({ type: 'metadata-file-parse-error' })
+  );
+});
+
+test.each<{ modifier: (castVoteRecordReportPath: string) => void }>([
```

Comment:
Without giving these a separate description, we get a bunch of identical test names:
<img width="673" alt="image" src="https://github.com/votingworks/vxsuite/assets/1938/bb9c0ef3-635c-4010-b522-faaee3b84266">

Reply:
Yeah I sometimes get lazy about this 😅

Updated such that all test descriptions are unique: https://github.com/votingworks/vxsuite/pull/4091/commits/af139cf547e8d8c23c224d779fe5f141406c0840